components/profileLink/index.jsx

```diff
@@ -37,34 +38,19 @@ export function buildProfileLink(webId, route, path) {
   return `${route}/${path}/${encodeURIComponent(webId)}`;
 }
 
-export default function ProfileLink(props) {
+export default function ProfileLink() {
   const { route } = useRouter();
-  const { webId } = props;
   const { thing } = useThing();
-  const contact = getProfileIriFromContactThing(thing);
-
-  const type = getUrl(thing, rdf.type);
-  const path = type === schema.Person ? "person" : "app";
+  const webId = getProfileIriFromContactThing(thing);
+  const profile = useFullProfile(webId);
+  const path = profile.types.includes(schema.Person) ? "person" : "app";
```

Comment:
On the first render of the `useFullProfile` hook, it returns `null`, as such you'd need to check if you've a `profile` value before looking at `profile.types.includes`

So you'll probably want a `if (profile === null) return null;` in here.

The comments below should probably also be removed: